src/main/java/uk/gov/hmcts/reform/sendletter/blob/BlobProcessor.java

```diff
@@ -16,28 +19,49 @@
 public class BlobProcessor {
     private static final Logger LOG = LoggerFactory.getLogger(BlobProcessor.class);
 
+    private final BlobManager blobManager;
     private final BlobReader blobReader;
     private final BlobBackup blobBackup;
     private final BlobStitch blobStitch;
     private final BlobDelete blobDelete;
+    private final LeaseClientProvider leaseClientProvider;
 
-    public BlobProcessor(BlobReader blobReader, BlobBackup blobBackup, BlobStitch blobStitch, BlobDelete blobDelete) {
+    public BlobProcessor(BlobReader blobReader, BlobBackup blobBackup, BlobStitch blobStitch,
+                         BlobDelete blobDelete, BlobManager blobManager,
+                         LeaseClientProvider leaseClientProvider) {
         this.blobReader = blobReader;
         this.blobBackup = blobBackup;
         this.blobStitch = blobStitch;
         this.blobDelete = blobDelete;
+        this.blobManager = blobManager;
+        this.leaseClientProvider =  leaseClientProvider;
     }
 
     public boolean read() throws IOException {
         LOG.info("BlobProcessor:: read blobs ");
         Optional<ManifestBlobInfo> blobInfo = blobReader.retrieveManifestsToProcess();
         if (blobInfo.isPresent()) {
-            var printResponse = blobBackup.backupBlobs(blobInfo.get());
-            LOG.info("BlobProcessor:: backup blobs response {}", printResponse);
-            LOG.info("BlobProcessor:: stitch blobs");
-            var deleteBlob = blobStitch.stitchBlobs(printResponse);
-            LOG.info("BlobProcessor:: delete original blobs");
-            //blobDelete.deleteOriginalBlobs(deleteBlob);
+
+            var containerClient  = blobManager.getContainerClient(blobInfo.get().getContainerName());
+            var blobClient = containerClient.getBlobClient(blobInfo.get().getBlobName());
+            var leaseClient = leaseClientProvider.get(blobClient);
+            try {
+                leaseClient.acquireLease(20);
```

Comment:
can we make this a class property? I assume it's seconds?

Reply:
sure